docs/userguide/format_parse/messages/try-mf2.md

```diff
@@ -0,0 +1,366 @@
+           .setPattern("Hello {$user}, today is {$now :date style=long}!", parseError, errorCode)
+           .setLocale(Locale("en_US"))
+           .build(errorCode);
+
+       std::map<UnicodeString, message2::Formattable> argsBuilder;
+       argsBuilder["user"] = message2::Formattable("John");
+       argsBuilder["now"] = message2::Formattable::forDate(date);
+       message2::MessageArguments arguments(argsBuilder, errorCode);
+
+       icu::UnicodeString result = mf.formatToString(arguments, errorCode);
+       std::string strResult;
+       result.toUTF8String(strResult);
+       std::cout << strResult << std::endl;
+    }
+    ```
+
+1. Build your application and run it
+    ```sh
+    g++ hello_mf2.cpp -I$DESTDIR/usr/local/include -std=c++17 -L$DESTDIR/usr/local/lib -licuuc -licudata -licui18n
+
+    # if macOS
+    DYLD_LIBRARY_PATH=$DESTDIR/usr/local/lib ./a.out
+    # else if Linux
+    LD_LIBRARY_PATH=$DESTDIR/usr/local/lib ./a.out
+    # end
+    ```
+
+1. This will output
+    ```
+    Hello John, today is January 13, 2025!
+    ```
+
+## C++ Windows with Visual Studio
+
+### From command line
+
+Start the Visual Studio "x64 Native Tools Command Prompt for VS 20xx"
+
+1. Prepare a sandbox folder
+
+    ```cmd
+    set ICU_SANDBOX=%USERPROFILE%\hello_icu_mf2
+    md %ICU_SANDBOX%
+    cd %ICU_SANDBOX%
+    ```
+
+1. Build ICU4C (you only need to do this once):
+
+    ```cmd
+    git clone https://github.com/unicode-org/icu.git
+    
+    cd icu\icu4c
+    msbuild source/allinone/allinone.sln /p:Configuration=Release /p:Platform=x64 /p:SkipUWP=true
+    cd ..\..
+    
+    set DESTDIR=%ICU_SANDBOX%\icu_release
+    rd /q/s %DESTDIR%
+    md %DESTDIR%
+    xcopy icu\icu4c\include %DESTDIR%\include /E /V /I /Q /Y
+    xcopy icu\icu4c\bin64   %DESTDIR%\bin64   /E /V /I /Q /Y
+    xcopy icu\icu4c\lib64   %DESTDIR%\lib64   /E /V /I /Q /Y
+    ```
+
+1. Create a minimal C++ file here (we are in the `$ICU_SANDBOX` folder).
+Let's call it `hello_mf2.cpp`.
+The content is listed above (see the Linux section).
+
+1. Build your application and run it
+
+    ```sh
+    g++ hello_mf2.cpp -I$DESTDIR/usr/local/include -std=c++17 -L$DESTDIR/usr/local/lib -licuuc -licudata -licui18n
+    
+    # if macOS
+    DYLD_LIBRARY_PATH=$DESTDIR/usr/local/lib ./a.out
+    # else if Linux
+    LD_LIBRARY_PATH=$DESTDIR/usr/local/lib ./a.out
+    # end
+    ```
+
+1. This will output
+    ```
+    Hello John, today is January 13, 2025!
+    ```
+
+### From Visual Studio (UI)
+
+1. Prepare a sandbox folder. Let's call it `hello_icu_mf2`.
+
+1. Build ICU4C (you only need to do this once)
+
+    * Clone the ICU repository from
+      <https://github.com/unicode-org/icu> to the `hello_icu_mf2` folder.
+      We will end up with the folder `hello_icu_mf2\icu`.
+
+        > :point_right: **Note**: You can also downloaded a pre-build artifact from [GitHub releases](https://github.com/unicode-org/icu/releases/tag/release-76-1). But in that case the 
```

Comment:
I don't think this note belongs here.
If we add it here it invalidates a most of the bullets after.

And we have a whole separate section with "download the pre-build artifacts" way of doing things.
We are now in the "build your own" mode.